controlplane/kubeadm/internal/control_plane.go

```diff
@@ -266,6 +268,8 @@ func (c *ControlPlane) UpToDateMachines() collections.Machines {
 		collections.Not(collections.ShouldRolloutAfter(&c.reconciliationTime, c.KCP.Spec.RolloutAfter)),
 		// Machines that match with KCP config.
 		MatchesMachineSpec(c.infraResources, c.kubeadmConfigs, c.KCP),
+		// Machines that shouldn't be rollout out if their certificates are not about to expire.
+		collections.Not(collections.ShouldRolloutBefore(&c.reconciliationTime, c.KCP.Spec.RolloutBefore)),
```

Comment:
nit: given that we moved rolloutbefore up in the func above, let's do the same here